src/ResourceManager.cpp

```diff
@@ -94,20 +161,27 @@ ResourceManager* ResourceManager::getInstance()
     return Base::Singleton<ResourceManager>::get();
 }
 
-Format::Dat::Item* ResourceManager::datFileItem(string filename)
+Format::Dat::Item* ResourceManager::datFileItem(const std::string& filename)
 {
-    std::transform(filename.begin(), filename.end(), filename.begin(), ::tolower);
-
     // Return item from cache
     auto itemIt = _datItemMap.find(filename);
     if (itemIt != _datItemMap.end())
     {
         return itemIt->second;
     }
 
+    std::string filenameLowered(filename);
```

Comment:
The old code does two things:
- Always makes a copy of an input string
- Transforms the string to lowercase

My point is that we can avoid these actions if input string already is in lowercase.
And then do the transformation only if we failed to find the input string.

If you think that we should use lowercased filenames only, then we can delete this step completely.